parallel_cluster/toil/README.md

```diff
@@ -29,19 +30,49 @@ mkdir -p "${TOIL_OUTPUTS}"
 ## Running TOIL
 
 ```bash
-srun 
+srun \
   --output "logs/toil.%j.%s.log" \
   --error "logs/toil.%j.%s.log" \
-  toil-cwl-runner \
-    --jobStore "${TOIL_JOB_STORE}.%s.log" \
-    --workDir "${TOIL_WORKDIR}" \
-    --outdir "${TOIL_OUTPUTS}" \
-    --writeLogs "${TOIL_LOG_DIR}" \
-    --batchSystem slurm \
-    my-cwl-tool.packed.json \
-    my-cwl-tool.packed.input.json
+  bash -c "\
+    toil-cwl-runner \
+      --jobStore \"${TOIL_JOB_STORE}/\${SLURM_JOB_ID}.\${SLURM_STEP_ID}.log\" \
+      --workDir \"${TOIL_WORKDIR}\" \
+      --outdir \"${TOIL_OUTPUTS}\" \
+      --writeLogs \"${TOIL_LOG_DIR}\" \
+      --batchSystem slurm \
+      --cleanWorkDir=onSuccess \
+      \"my-cwl-tool.packed.json\" \
+      \"my-cwl-tool.packed.input.json\""
 ```
 
+## Where are all my files
+
+Before traversing the many files inside `TOIL_JOB_STORE` or `TOIL_WORKDIR`, it may be of consideration to
+look at `logs/toil.*.*.log` which will print the command that has failed.  
+If you think this command is correct and you need to debug further, I would recommend tinkering the `--cleanWorkDir`
+parameter of the command above. I've changed this from `always` to `onSuccess` meaning all failed jobs will be 
+able to investigated
+
+## Why is the job file in binary format.
+
+This is a great question. If you would like to look at the job file you can use some python magic 
+to look deeper into these job files
+
+```python
+from pprint import pprint
+import pickle
+with open("/path/to/binary/job", "rb") as pickle_h:
+  job_obj = pickle.load(pickle_h)
+pprint(job_obj)
+```
+
+If you have a look at the `job_obj` attribute 'command' this should lead you to a 'stream' object.
+You may use the same method above to import the stream file which may assist you further in your debugging.
+
+## Rerunning a workflow
+I would recommend ensuring that no docker containers are continuing to run in the backgroud before a workflow 
```

Comment:
Bummer, can you put a link to that bug report/issue?

Reply:
I think it's a slurm thing rather than a toil problem. Like slurm isn't keeping track of the processes launched.